### PR TITLE
Commands to easily manage deletion of history from the current session.

### PR DESCRIPTION
**Pull request by Eadaen**
Solves #3033.

Provides `history clear` which purges past commands from the current session; `history off`, which stops any history from being saved from the entire current session; and `history on`, which re-enables history saving in the current session. 

Works for both JSON and SQLite backends, and it works on both the history buffer and on history already saved to disk.

Builds off of #3691. Thank you @Itay4 for starting out.

### CODE REVIEW

File: xonsh/history/json.py [hunk 1]
```diff
@@ -502,3 +507,22 @@ def run_gc(self, size=None, blocking=True, force=False):
         if blocking:
             while self.gc.is_alive():  # while waiting for gc.
                 time.sleep(0.1)  # don't monopolize the thread (or Python GIL?)
+
+    def clear(self):
+        """Clears the current session's history from both memory and disk."""
+
+        def wipe_memory(hist):
```
Comment (scopatz) [inline xonsh/history/json.py:514]: This probably doesn't need to be a function

**Comment by scopatz** [comment]
Thanks for putting this in @Eadaen

File: xonsh/history/base.py [hunk 2]
```diff
@@ -153,3 +154,21 @@ def run_gc(self, size=None, blocking=True):
             If set blocking, then wait until gc action finished.
         """
         pass
+
+    @staticmethod
+    def remember_history_check(f):
```
Comment (scopatz) [inline xonsh/history/base.py:159]: Why does this decorator need to exist? It seems easier/faster to put the logic in the actual functions where it is needed. All this is doing is a simple if-condition.

Comment (Eadaen) [reply xonsh/history/base.py:159]: It doesn't need to exist. In my opinion it's cleaner/more readable to use the decorator, but I'm happy to move the check into the individual functions if that's what you prefer.

Comment (scopatz) [reply xonsh/history/base.py:159]: Yeah, please move them. I am pretty sure that this implementation is slower.

**Comment by scopatz** [comment]
Thanks @Eadaen - would you mind running black?

**Comment by Eadaen** [comment]
Ignore the last commit. I didn't realise you'd already run Black on json.py in the master branch.

**Comment by scopatz** [comment]
Looks like there is a conflict here now. Sorry for the back and forth @Eadaen

**Comment by Eadaen** [comment]
I think that should fix the conflict. I wasn't sure the best way to revert the problem commit.

No problem @scopatz. Thanks for making sure everything is correct.

**Comment by scopatz** [comment]
Thanks!

**Pull request merged**